### PR TITLE
add crypto/secp256k1  recovering pubkey from signature

### PR DESCRIPTION
**Pull request by wupeaking**
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by Liamsi** [comment]
Ref: https://github.com/tendermint/tendermint/issues/3413

File: crypto/secp256k1/secp256k1_cgo.go [hunk 1]
```diff
@@ -19,5 +20,33 @@ func (privKey PrivKeySecp256k1) Sign(msg []byte) ([]byte, error) {
...
+	if len(sig) == 65 {
+		sig = sig[0:64]
+	}
 	return secp256k1.VerifySignature(pubKey[:], crypto.Sha256(msg), sig)
 }
+
+// SignRecoverAble creates an ECDSA signature on curve Secp256k1, a recoverable ECDSA signature .
+// it returns the result that is in 65-byte [R || S || V] (Ethereum signature format)
+func (privKey PrivKeySecp256k1) SignRecoverAble(msg []byte) ([]byte, error) {
+	rsv, err := secp256k1.Sign(crypto.Sha256(msg), privKey[:])
```
Comment (Liamsi) [inline crypto/secp256k1/secp256k1_cgo.go]: this could be simplified to `returned secp256k1.Sign(crypto.Sha256(msg), privKey[:])`

**Comment by Liamsi** [comment]
`TestListenerTimeoutReadWrite` failed. Seems unrelated to these changes but seems like a new non-deterministic error: 
```
--- FAIL: TestListenerTimeoutReadWrite (0.01s)
    socket_listeners_test.go:110: read tcp 127.0.0.1:41375->127.0.0.1:56014: i/o timeout
```
https://circleci.com/gh/tendermint/tendermint/50585#tests/containers/1
restarted the build and will open an issue.

**Comment by wupeaking** [comment]
ok, I am very happy to submit a PR for this project. Very interested in this project.  Do you need me to temporarily close this issue? ^_^

**Comment by Liamsi** [comment]
Yes, let's close this for now. Thanks again! 🚀

**Pull request closed without merging**